src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/controller/RefreshControllerIntegrationTest.java

```diff
@@ -0,0 +1,340 @@
+
+    private static final Logger logger = LoggerFactory.getLogger(RefreshControllerIntegrationTest.class);
+
+    private static final String REFRESH_JOB_RECORDS_QUERY = "SELECT job_id, status, user_ids, linked_job_id,"
+            + " comments, log FROM refresh_jobs where job_id=?";
+    private static final String AUTHORISED_SERVICE = "ccd_gw";
+    private static final String ROLE_NAME_STCW = "senior-tribunal-caseworker";
+    private static final String ROLE_NAME_TCW = "tribunal-caseworker";
+    private static final String URL = "/am/role-mapping/refresh";
+
+    private MockMvc mockMvc;
+    private JdbcTemplate template;
+
+    @Inject
+    private WebApplicationContext wac;
+
+    @Autowired
+    private DataSource ds;
+
+    @MockBean
+    private CRDFeignClient crdFeignClient;
+
+    @MockBean
+    private RequestMappingService requestMappingService;
+
+    @MockBean
+    private TopicConsumer topicConsumer;
+
+    private static final MediaType JSON_CONTENT_TYPE = new MediaType(
+            MediaType.APPLICATION_JSON.getType(),
+            MediaType.APPLICATION_JSON.getSubtype(),
+            StandardCharsets.UTF_8
+    );
+
+    @Before
+    public void setUp() {
+        template = new JdbcTemplate(ds);
+        mockMvc = MockMvcBuilders.webAppContextSetup(wac).build();
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = {"classpath:sql/insert_refresh_jobs.sql"})
+    public void shouldProcessRefreshRoleAssignmentsWithJobIdToComplete() throws Exception {
+        logger.info(" RefreshJob record With Only JobId to process successful");
+        Long jobId = 1L;
+        RefreshJobEntity refreshJob = getRecordsFromRefreshJobTable(jobId);
+        assertEquals("NEW", refreshJob.getStatus());
+
+        mockCRDService();
+        mockRequestMappingServiceWithStatus(HttpStatus.CREATED);
+
+        mockMvc.perform(post(URL)
+                .contentType(JSON_CONTENT_TYPE)
+                .headers(getHttpHeaders())
+                .param("jobId", jobId.toString()))
+                .andExpect(status().is(202))
+                .andReturn();
+
+        Thread.sleep(1000);
+        logger.info(" -- Refresh Role Assignment record updated successfully -- ");
+        refreshJob = getRecordsFromRefreshJobTable(jobId);
+        assertEquals(COMPLETED, refreshJob.getStatus());
+        assertEquals(0, refreshJob.getUserIds().length);
+        assertNotNull(refreshJob.getLog());
+    }
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = {"classpath:sql/insert_refresh_jobs.sql"})
+    public void shouldProcessRefreshRoleAssignmentsWithJobIdToAborted() throws Exception {
+        logger.info(" RefreshJob record With Only JobId to process Aborted");
+        Long jobId = 1L;
+
+        mockCRDService();
+        mockRequestMappingServiceWithStatus(NOT_ACCEPTABLE);
```

Comment:
We need to check 2 types of scenarios one is recoverable error which is 5XX and another non recoverable like 422. please evaluate 5xx here and add a new scenario for specifically for 422.